zerver/lib/streams.py

```diff
@@ -642,3 +642,20 @@ def get_stream_by_narrow_operand_access_unchecked(operand: Union[str, int], real
     if isinstance(operand, str):
         return get_stream(operand, realm)
     return get_stream_by_id_in_realm(operand, realm)
+
+
+def get_stream_admin_dict_for_streams(stream_ids: List[int]) -> Dict[int, List[int]]:
```

Comment:
It might be cleaner for this to accept a list of `recipient_id`s just to make the query below cleaner, in combination with the more natural `id, role, active` database index we'd want to add.